infra/elasticsearch/src/types/Document.ts

```diff
@@ -0,0 +1,129 @@
+/* eslint-disable @typescript-eslint/naming-convention */
+
+export interface Document {
```

Comment:
I'm not a huge fan of a big flat document. Here, the action is a no-nullish string but for a code-du-travail doc, I have this:
```json
{
		"url": "https://www.legifrance.gouv.fr/affichCodeArticle.do;?idArticle=LEGIARTI000033333034&cidTexte=LEGITEXT000006072050",
		"html": "<p>Les recours hiérarchiques dirigés contre les décisions mentionnées aux articles <a href='/affichCodeArticle.do?cidTexte=LEGITEXT000006072050&idArticle=LEGIARTI000018495600&dateTexte=&categorieLien=cid'>R. 5221-17</a>, <a href='/affichCodeArticle.do?cidTexte=LEGITEXT000006072050&idArticle=LEGIARTI000018495642&dateTexte=&categorieLien=cid'>R. 5221-32 et suivants </a>sont formés auprès du ministre chargé de l'immigration.</p>",
		"dateDebut": 1477958400000,
		"description": "Les recours hiérarchiques dirigés contre les décisions mentionnées aux articles R. 5221-17 , R. 5221-32 et suivants sont formés auprès du ministre chargé de l'immigration",
		"breadcrumbs": [],
		"cdtnId": "232a43a527",
		"excludeFromSearch": false,
		"id": "LEGIARTI000018495604",
		"isPublished": true,
		"metaDescription": "Les recours hiérarchiques dirigés contre les décisions mentionnées aux articles R. 5221-17 , R. 5221-32 et suivants sont formés auprès du ministre chargé de l'immigration",
		"slug": "r5221-19",
		"source": "code_du_travail",
		"text": "Les recours hiérarchiques dirigés contre les décisions mentionnées aux articles R. 5221-17 , R. 5221-32 et suivants sont formés auprès du ministre chargé de l'immigration.\n",
		"title": "R5221-19"
	}
```
There is no action. So action can be Nullable.

I think we can split by document source, something like that:
```ts 
interface FicheServicePublic {
   // FicheServicePublic specific fields
    data: string;
}

interface Contribution {
   // Contribution specific fields
    legalReferences: string;
}

enum Source {
    CONTRIBUTION = "contribution",
    FICHE_SP = "fiche_service_public",
} 

type Doc = {
   // Common fields 
    cdtn_id: string;
    source: Source;
} & (FicheServicePublic | Contribution)

const data: Doc[] = [{
    "cdtn_id": "ID1",
    "source": "fiche_service_public",
    "data": "La donnée"
}, {
    "cdtn_id": "ID2",
    "source": "contribution",
    "legalReferences": "Références légales"
}];

data.forEach((item) => {
    console.log("ID: ", item.cdtn_id);
    switch (true) {
        case item.source === Source.FICHE_SP:
        console.log("Item: ", (item as FicheServicePublic).data);
        break;
        case item.source === Source.CONTRIBUTION:
        console.log("Item: ", (item as Contribution).legalReferences);
        break;
    }
})
```